mbedtls/src/ssl/context.rs

```diff
@@ -149,7 +149,7 @@ impl Context {
         if let Some(s) = hostname {
             let cstr = alloc::ffi::CString::new(s).map_err(|_| Error::SslBadInputData)?;
             unsafe {
-                ssl_set_hostname(self.into(), cstr.as_ptr())
+                ssl_set_hostname(self.into(), cstr.as_ptr() as *const c_char)
```

Comment:
does this actually fix the problem?
Looking at https://doc.rust-lang.org/std/ffi/struct.CStr.html#method.as_ptr

> The type of the returned pointer is [*const c_char](https://doc.rust-lang.org/std/ffi/type.c_char.html), and whether it’s an alias for *const i8 or *const u8 is platform-specific.

From that description It seems that it might be a mismatch between the rust and the c compiler flags on one of the platforms

Reply:
per discussion it's due to, https://github.com/mobilecoinfoundation/rust-mbedtls/blob/mc-develop/mbedtls-sys/src/types.rs#L45
which isn't coming from std